tools/tools-sync-renv.R

```diff
@@ -1,3 +1,8 @@
+# This script updates the vendored version of renv. It must be run from the
+# Packrat root directory. By default, uses the current version of renv on main.
+# You can pass a single tag as an argument after --args to use that version.
+# R -f /tools/test.R --args 0.15.5-48
```

Comment:
Are renv tags ever overwritten? It appears as if @kevinushey generally bumps the version, but I don't know if that's true of every commit.

Should we suggest using commit references rather than tags?

Reply:
I'll see what it would be like to use commit references.

Reply:
To do this would require changing the conditional flow to the following, and would only accept *full shas*, not abbreviations.

```r
if (is.null(ref)) {
  # check out the latest renv sources
  system("git clone --depth 1 https://github.com/rstudio/renv")
  twd <- setwd("renv")
} else if (length(ref) == 1) {
  # check out the tagged version
  dir.create("renv")
  twd <- setwd("renv")
  system("git init")
  system("git remote add origin https://github.com/rstudio/renv")
  system(paste("git fetch --depth 1 origin", ref))
  system("git checkout FETCH_HEAD")
}

gitCommitHash <- system("git rev-parse HEAD", intern = TRUE)
setwd(twd)
```

I'm going to leave this here for now, and not make this change unless you feel strongly that this is better. I feel comfortable treating tags on `renv` as reasonable ways to find commits, and as you wrote in another comment, the header comment in the resulting `renv.R` file mentions the commit that was used, so… I don't _think_ this causes too much of a problem.